state-and-event-handling/event-handling.md

```diff
@@ -0,0 +1,182 @@
+
+Consider this example JSX, where a `SomeElement` element will listen for click events.
+
+<!-- prettier-ignore-start -->
+```js
+<SomeElement onClick={...}></SomeElement>
+```
+<!-- prettier-ignore-end -->
+
+## Event Handling Functions
+
+The value of our event listener should be a function.
+
+This function will be invoked whenever the appropriate event occurs.
+
+Consider this `Post` component. It defines a function named `printMessage`. Then, it returns JSX that includes a "like" button.
+
+The "like" button listens for click events. By setting the value of `onClick` to `printMessage`, our web app invokes `printMessage` whenever we click the like button.
+
+<!-- prettier-ignore-start -->
+```js
+const Post = () => {
+    const printMessage = () => {
+        console.log('Hello! We\'re in printMessage!');
+    }
+
+    return (<section>
+        <button onClick={printMessage}>Like</button>
+    </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+![An app with a "like" button, and browser Dev Tools that read "Hello! We're in printMessage!" in the console.](../assets/state-and-event-handling_event-handling_printMessage.png)  
+_Fig. An app with a "like" button, and browser Dev Tools that read "Hello! We're in printMessage!" in the console._
+
+Every time we click our "Like" button, we'll see our message in the console!
+
+```
+Hello! We're in printMessage!
+```
+
+### Event Handling Functions Can Accept `event`
+
+We can get details about our events in our event-handling functions.
+
+If our event-handling function accepts arguments, the first argument will be an event object.
+
+Consider this code, where `printMessage` accepts an event object, and then prints it.
+
+<!-- prettier-ignore-start -->
+```js
+const Post = () => {
+    const printMessage = (event) => {
+        console.log('Hello! We\'re in printMessage!');
+        console.log('This event object contains details about the event:', event);
+    }
+
+    return (<section>
+        <button onClick={printMessage}>Like</button>
+    </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+This is what prints to the browser console. (Your results may vary depending on your browser.)
+
+```
+Hello! We're in printMessage!
+This event object contains details about the event: 
+Object { _reactName: "onClick", ...
+```
+
+### Event Handling Functions Are Functions
+
+It's helpful to know that our event handling functions are, otherwise, normal functions.
+
+They can call other functions inside them, and pass arguments to them, too.
+
+Consider this `Post` component. It:
+
+- Defines a function, `printName`, which takes in a `name` argument
+- Defines another function, `printMessage`. It calls `printName` with the argument `"Ada Lovelace"`.
+- Sets `printMessage` as the click event handler for a "Like" button
+
+<!-- prettier-ignore-start -->
+```js
```

Comment:
I really want to bring up closure binding again. I think it's ok to skip it here, since if we squint at this, we can convince ourselves that we're just looking at a funny looking class. But eventually closure semantics wind up rearing their head and ordering can become an issue (especially with useCallback dependencies).

Reply:
YEAHHHHHH, I just finished writing the `lifting-state-up` PRs and I really craved this.

I'll copy/paste this into Slack for a fun convo, buuUUUT I had made the call that:

- Closures are a lightning talk topic
- Specifically for C15, I know that one of the Closure lightning talk talks with be late 🙈 Not ideal...

Reply:
I really don't want to forget this, so I made an issue for it to help me remember: https://github.com/Ada-Developers-Academy/core-unit-3/issues/44